### PR TITLE
!!! TASK: Move workspace module to separate package

### PR DESCRIPTION
**Pull request by ahaeslich**
**Tasks**

- [x] Create new package named "Neos.Workspace.Ui"
- [x] Move settings
- [x] Move translations
- [x] Only minimal adjustments in controller code
- [x] Use singular for controller name
- [ ] Register package in neos distribution
- [x] Create core migration to adjust existing privilege usages ❓

**Upgrade instructions**

Add `neos/workspace-ui` as dependency.

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Workspace.Ui/Classes/Controller/WorkspacesController.php
Comment (bwaidelich) [inline Neos.Workspace.Ui/Classes/Controller/WorkspacesController.php:1]: This controller should have been named `WorkspaceController` (singular) – maybe moving it is a chance to fix that!?

**Review by kdambekalns — APPROVED** [review]
Seems fine by reading…

File: Neos.Workspace.Ui/Classes/Controller/WorkspaceController.php [hunk 1]
```diff
@@ -502,7 +499,7 @@ public function rebaseAndRedirectAction(Node $targetNode, Workspace $targetWorks
         $this->redirect(
             'show',
             'Frontend\\Node',
-            'Neos.Neos',
+            'Neos.Workspace.Ui',
```
Comment (mhsdesign) [inline Neos.Workspace.Ui/Classes/Controller/WorkspaceController.php:502]: fyi here it must stay `Neos.Neos` but ill fix that

**Comment by mhsdesign** [comment]
Aaaand if we have a list for creating new packages we should also add that we need to adjust to include it in phpstan as well.

Btw did you migrate all the translation files by hand oO soooo much work :O or can weblate do this as well? xD

**Comment by Sebobo** [comment]
Did it by hand, while you were talking during the sprint 😉

**Comment by kdambekalns** [comment]
> Did it by hand, while you were talking during the sprint 😉

I _think_ it would have been enough to do it in the `en` files, Weblate _should_ clean things up as needed then…

**Comment by dlubitz** [comment]
It seems, there are references to the moved translations, which where not updated and are not working anymore.

E.g: https://github.com/neos/neos-development-collection/blob/9.0/Neos.Media.Browser/Resources/Private/Templates/Usage/RelatedNodes.html#L127-L127

Are there any plans to find and change them? 

(I fixed them in the RelatedNodes.html, but I guess there are more in other templates)

**Comment by ahaeslich** [comment]
Thx for finding the issue. 

> Are there any plans to find and change them?

It seems we have to check more views for this labels.

**Comment by dlubitz** [comment]
We might re-think which package owns the translations. E.g. the translation of "Workspace" should not be owned by a workspace package. As this is a general translation in Neos. Otherwise we would need to have a dependency to the workspace package in each package where we use the "Workspace" translation.

**Comment by dlubitz** [comment]
E.g. The Media Browser... By using the translation keys of Workspace.Ui I made the Media Browser package dependent on the Workspace UI package. Which shouldn't be.

**Pull request merged**